cpp/src/arrow/flight/server.h

```diff
@@ -200,10 +201,11 @@ class ARROW_FLIGHT_EXPORT FlightServerBase {
   int GotSignal() const;
 
   /// \brief Shut down the server. Can be called from signal handler or another
-  /// thread while Serve() blocks.
+  /// thread while Serve() blocks. Optionally a deadline can be set. Once the
+  /// the deadline expires all pending calls associated with the server will be
+  /// forcefully canceled.
```

Comment:
Because gRPC won't cancel pending _sync_ calls which is what Flight uses, this is a little misleading. (See docs: https://grpc.github.io/grpc/cpp/classgrpc_1_1_server_interface.html#a644772d11101318f1424e595dee73ccf)

Reply:
```suggestion
  /// thread while Serve() blocks. Optionally a deadline can be set. Once the
  /// the deadline expires server will switch to shutdown mode. All pending
  /// async calls associated with the server will be forcefully canceled.
  /// Remaining running sync calls will block until complete.
```

Reply:
Does this look ok @lidavidm ? I'm not sure if flight uses the sync/async language.

Reply:
Ah, sorry, Flight does not use async gRPC at all. So what I mean is basically, "even if the deadline expires, this method will wait for existing calls to finish", or "the deadline does nothing" (right now :slightly_smiling_face:)

Reply:
Got it :). So how about:
```suggestion
  /// thread while Serve() blocks. Optionally a deadline can be set. Once the
  /// the deadline expires server will wait until remaining running calls 
  /// complete.
```

Reply:
Sounds good to me!

Reply:
Done!